File: 3277-maximum-xor-score-subarray-queries/3277-maximum-xor-score-subarray-queries.java
class Solution {
		public int[] maximumSubarrayXor(int[] nums, int[][] queries) {
			int n = nums.length;
			int[][] t = new int[n][n];
			for(int j = 0;j < n;j++)t[j][j] = nums[j];

			for(int j = 1;j < n;j++){
				for(int i = 0;i+j < n;i++){
					t[i][i+j] = t[i][i+j-1] ^ t[i+1][i+j];
				}
			}

			for(int j = 1;j < n;j++){
				for(int i = 0;i+j < n;i++){
					t[i][i+j] = Math.max(t[i][i+j], Math.max(t[i][i+j-1], t[i+1][i+j]));
				}
			}

			int[] ans = new int[queries.length];
			for(int i = 0;i < queries.length;i++){
				int l = queries[i][0];
				int r = queries[i][1];
				ans[i] = t[l][r];
			}
			return ans;
		}
	}

File: 3277-maximum-xor-score-subarray-queries/3277-maximum-xor-score-subarray-queries.cpp
class Solution {
public:
    vector<int> maximumSubarrayXor(vector<int>& nums, vector<vector<int>>& queries) {
        int n = nums.size(), q = queries.size();

        vector<vector<int>> dp(n, vector<int>(n));
        for(int i = n - 1; ~i; i--)
        for(int j = i; j < n; j++){
            int x = nums[i], z = j - i;
            for(int l = z; l; l = (l - 1) & z){
                x ^= nums[i + l];
            }
            dp[i][j] = x;
            if(i < j){
                dp[i][j] = max({dp[i][j], dp[i + 1][j], dp[i][j - 1]});
            }
        }

        vector<int> ret;
        for(auto &qry : queries){
            int x = qry[0], y = qry[1];
            ret.push_back(dp[x][y]);
        }
        return ret;
    }
};
